esp-hal/src/macros.rs

```diff
@@ -168,3 +168,71 @@ macro_rules! if_set {
 macro_rules! ignore {
     ($($item:tt)*) => {};
 }
+
+/// Macro to define the application descriptor.
+#[macro_export]
+macro_rules! esp_app_desc {
+    () => {
+        #[repr(C)]
+        pub struct EspAppDesc {
+            pub magic_word: u32,     // Magic word ESP_APP_DESC_MAGIC_WORD
+            pub secure_version: u32, // Secure version
+            pub reserv1: [u32; 2],   // reserv1
+            pub version: [core::ffi::c_char; 32], // Application version
+            pub project_name: [core::ffi::c_char; 32], // Project name
+            pub time: [core::ffi::c_char; 16], // Compile time
+            pub date: [core::ffi::c_char; 16], // Compile date
+            pub idf_ver: [core::ffi::c_char; 32], // Version IDF
+            pub app_elf_sha256: [u8; 32], // sha256 of elf file
+            pub min_efuse_blk_rev_full: u16, // Minimal eFuse block revision supported by image
+            pub max_efuse_blk_rev_full: u16, // Maximal eFuse block revision supported by image
+            pub mmu_page_size: u8,   // MMU page size in log base 2 format
+            pub reserv3: [u8; 3],    // reserv3
+            pub reserv2: [u32; 18],  // reserv2
+        }
+
+        #[no_mangle]
+        #[used]
+        #[link_section = ".rodata_desc"]
+        #[allow(non_upper_case_globals)]
+        pub static esp_app_desc: EspAppDesc = {
+            const ESP_APP_DESC_MAGIC_WORD: u32 = 0xABCD5432;
```

Comment:
Will this ever change? I.e if we update the bootloaders to some future esp-idf v6 version, do they change this magic value?

Reply:
the oldest commit I found in esp-idf on GitHub is from Aug 17, 2022 and it was the same back then

I _guess_ it should only change if the descriptor changes in an incompatible way and if it breaks then it's a good thing - so it's apparently not corresponding to esp-idf versions